serdes/serde-common/src/main/java/io/apicurio/registry/serdes/SchemaResolver.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2021 Red Hat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.registry.serdes;
+
+import java.util.Map;
+
+import org.apache.kafka.common.header.Headers;
+
+import io.apicurio.registry.rest.client.RegistryClient;
+import io.apicurio.registry.serdes.strategy.ArtifactIdStrategy;
+
+/**
+ * @author Fabian Martinez
+ */
+public interface SchemaResolver<SCHEMA, DATA> {
+
+    /**
+     * Configure, if supported.
+     *
+     * @param configs the configs
+     * @param isKey are we handling key or value
+     */
+    default void configure(Map<String, ?> configs, boolean isKey, SchemaMapper<SCHEMA, DATA> schemaMapper) {
+    }
+
+    public void setClient(RegistryClient client);
+
+    public void setArtifactIdStrategy(ArtifactIdStrategy<SCHEMA> artifactIdStrategy);
```

Comment:
I'm not sure if this should be a strategy still.  Why not just let this be another in-built function of the `SchemaResolver`?  Either way, this needs to be expanded to be some sort of `ArtifactCoordinatesStrategy` so that we can map both the groupId and artifactId.